back-end/aws-cloud9-api/src/main/java/com/serverless/dal/Question.java

```diff
@@ -0,0 +1,246 @@
+    	this.type = body.get("type").asText();
+    	this.question = body.get("name").asText();
+    	this.answer = body.get("answer").asInt();
+    	this.recruiter_id = body.get("recruiter_id").asText();
+    	this.language = body.get("language").asInt();
+    	this.form_membership = body.get("form_id").asText();
+    }
+    	
+    if (body.get("type").asText().equals("1")) {
+    	this.id = body.get("id").asText();
+    	this.type = body.get("type").asText();
+    	this.question = body.get("name").asText();
+    	this.answer = body.get("answer").asInt();
+    	this.recruiter_id = body.get("recruiter_id").asText();
+    	this.language = body.get("language").asInt();
+    	this.form_membership = body.get("form_id").asText();
+    	String answers = body.get("answer_list").toString(); //do zmiany
+    	logger.info("answers in string"+answers);
+    	this.answer_list =QuestionDBTable.singleChars(answers);
+    	logger.info("answers in answer_list"+answer_list);  	
+    }
+    	
+    if (body.get("type").asText().equals("2")) {
+    	this.id = body.get("id").asText();
+    	this.type = body.get("type").asText();
+    	this.question = body.get("name").asText();
+    	this.recruiter_id = body.get("recruiter_id").asText();
+    	this.language = body.get("language").asInt();
+    	this.form_membership = body.get("form_id").asText();
+
+    }
+    	
+    };
+
+   
+    public void update(JsonNode body) {
+		 logger.info("Call Question::update(" + body + ")");
+		 JsonNode newId = body.get("id");
+		 JsonNode newType = body.get("type");
+		 JsonNode newQuestion = body.get("name");
+		 JsonNode newAnswer = body.get("answer");
+		 JsonNode newRecrutierId = body.get("recruiter_id");
+		 JsonNode newForm_membership = body.get("form_membership");
+		 JsonNode newLanguage = body.get("language");
+		 JsonNode newAnswer_list = body.get("answer_list");
+
+		 
+		 
+		 
+		 if(newId != null) {
+			 this.setId(newId.asText());
+		 }
+		 else {
+			 logger.warn("Id is null");
+		 }
+		 
+		 if(newType != null) {
+			 this.setType(newType.asText());
+		 }
+		 else {
+			 logger.warn("newType is null");
+		 }
+		 
+		 if(newQuestion != null) {
+			 this.setQuestion((newQuestion.asText()));
+		 }
+		 else {
+			 logger.warn("Quesrtion is null");
+		 }
+		 if(newRecrutierId != null) {
+			 this.setRecruiter_id((newRecrutierId.asText()));
+		 }
+		 else {
+			 logger.warn("recruiterid is null");
+		 }
+		 if(newForm_membership != null) {
+			 this.setForm_membership((newForm_membership.asText()));
+		 }
+		 else {
+			 logger.warn("membership is null");
+		 }
+		 if(newLanguage != null) {
+			 this.setLanguage((newLanguage.asInt()));
+		 }
+		 else {
+			 logger.warn("Language is null");
+		 }
+		 if(newAnswer != null) {
+			 this.setAnswer((newAnswer.asInt()));
+		 }
+		 else {
+			 logger.warn("Answer is null");
+		 }
+		 if(newAnswer_list != null) {
+			 this.setAnswer((newAnswer_list.asInt()));
+		 }
+		 else {
+			 logger.warn("Answer is null");
+		 }
+	 }
```

Comment:
Za dużo tu tego. Wg wymagań funkcja może mieć max 50 linii. Może to sprawdzenie czy jest null zrobić w setterach, a tu po prostu po kolei wywoływać te settery?

Reply:
jak nie seterów wygenerowanych automatycznie tylko coś się w nich dłubie to wywala błędy, Michał miał to samo jak próbował własne settery robić